src/app/app.component.html

```diff
@@ -10,10 +10,10 @@
             <li *ngIf="auth.authenticated()">
                 <a materialize="dropdown" class="dropdown-button" data-beloworigin="true" data-hover="true" href="#!" data-activates="dropdownproject">{{auth.getCurrentDisplayName()}}<i class="material-icons right">arrow_drop_down</i></a>
                 <ul id="dropdownproject" class="dropdown-content">
-                    <li><a routerLink="/account/{{auth.getCurrentUserId()}}" routerLinkActive="active" class="black-text text-lighten-2">My Account</a></li>
-                    <li  *ngIf="auth.isVolunteer()"><a routerLink="/profile/{{auth.getCurrentUserId()}}" routerLinkActive="active" class="black-text text-lighten-2">My Profile</a></li>
+                    <li><a routerLink="/account/{{currentUserId}}" routerLinkActive="active" class="black-text text-lighten-2">My Account</a></li>
```

Comment:
:+1: that's fine. as a resolution, i have researched into this bug and although it's claimed to be fixed occasionally, i have not found in my experience for `null` to not crash the router. as a "fix", i use `val || safeValue`, where safeValue is usually 0 or 'asdf', so that the router doesn't crash.